### ISSUE TITLE
is any plan to support RxJava3?

### ISSUE THREAD

**Issue by Dugin123** [airbnb/MvRx#352]
is any plan to support RxJava3?

**Comment by gpeal** [comment]
@BenSchwab @elihart This might be a good opportunity to the coroutines defualt + rxjava2  and rxjava3 extension libraries. What do you think?

**Comment by Dugin123** [comment]
I prefer RxJava than coroutines. I think RxJava is more powerful and fitable functional programming.
  what do you think below structure? 
core+rxjava3
core+rxjava2
core+coroutines?

Anyway, regardless of module structure, RxJava3 has released official ver, so I hope it will be support.

**Comment by elihart** [comment]
Yes, I think this would tie in nicely to split extension libraries for these options. I would think that an extension library for rxjava3 would be pretty simple once we have one for rxjava2.

**Comment by jpventura** [comment]
@Dugin123 RxJava 2.x officially is [at maintenance mode](https://github.com/ReactiveX/RxJava/wiki/What's-different-in-3.0#2x-now-in-maintenance-mode).

Thus we could piggyback coroutines refactor to simply offer 3.x built over extensions when [JetBrains officially support it](https://github.com/Kotlin/kotlinx.coroutines/tree/master/reactive/kotlinx-coroutines-rx2)  (see #347).

**Comment by Dugin123** [comment]
@jpventura 
Now JetBrains  support [kotlinx-coroutines-rx3](https://github.com/Kotlin/kotlinx.coroutines/tree/master/reactive/kotlinx-coroutines-rx3).

**Comment by renaudcerrato** [comment]
RxJava2 is in maintenance mode and support will stop on March 1st 2021 ([source](https://github.com/ReactiveX/RxJava#version-2x)). Migrating is easy, will you accept a PR ?

**Comment by elihart** [comment]
Yes, if you look at the 2.0 release branch (https://github.com/airbnb/MvRx/tree/release/2.0.0) there is a `mvrx-rxjava2` module. I think we would be open to a PR that adds an equivalent module for rxjava3. I imagine it would be pretty straightforward to mirror what is done in the rxjava2 module

**Comment by JagadeeshDamarasingu** [comment]
hi @gpeal @elihart , I have opened a new PR #685 for this issue, please review
